samples: download: Move to `net/` and add nRF7002 support (#11547)
nrfconnect/sdk-nrf: samples/net/download/README.rst

The sample first initializes the :ref:`nrfxlib:nrf_modem` and AT communications.
Next, it provisions a certificate to the modem using the :ref:`modem_key_mgmt` library if the :ref:`CONFIG_SAMPLE_SECURE_SOCKET <CONFIG_SAMPLE_SECURE_SOCKET>` option is set.

> When compiling for an nRF91 series device, the provisioning of the certificates must be done before connecting to the LTE network since the certificates can only be provisioned when the device is not connected.

Reviewer: ```suggestion
When using an nRF91 Series device, the provisioning of the certificates must be done before connecting to the LTE network since the certificates can only be provisioned when the device is not connected.
```

Using TLS and DTLS
==================

> By default, :ref:`CONFIG_SAMPLE_SECURE_SOCKET <CONFIG_SAMPLE_SECURE_SOCKET>` option is set, which means that the sample provisions the certificate found in the :file:`samples/net/download/cert` folder.

Reviewer: ```suggestion
By default, the :ref:`CONFIG_SAMPLE_SECURE_SOCKET <CONFIG_SAMPLE_SECURE_SOCKET>` option is set, which means that the sample provisions the certificate found in the :file:`samples/net/download/cert` folder.
```

   Download client sample started
   Provisioning certificate
   Connecting to network
   IP Up
   Network connected
   Downloading https://nrfconnectsdk.s3.eu-central-1.amazonaws.com/sample-img-100kb.png
   [ 100% ] |==================================================| (102923/102923 bytes)
   Download completed in 13679 ms @ 7524 bytes per sec, total 102923 bytes
   IP down
   Disconnected from network
   Socket closed
   Bye

Dependencies
************

> This sample uses the following |NCS| libraries when compiled for nRF91 series devices:

Reviewer: ```suggestion
This sample uses the following |NCS| libraries when using an nRF91 Series device:
```
